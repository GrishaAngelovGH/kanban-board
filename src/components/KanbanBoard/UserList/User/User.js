import { useState } from "react"

import Avatar from "components/Avatar"

const User = ({ id, name, image, selected, onSelect }) => {
  const [isSelected, setIsSelected] = useState(selected)

  const handleToggle = () => {
    setIsSelected(!isSelected)
    onSelect(!isSelected, id)
  }

  return (
    <div className="row align-items-center" onClick={handleToggle}>
      <div className="col-2">
        <Avatar user={{ image, name }} />
      </div>
      <div className="col-8">
        <p className="m-0 fw-bold">{name}</p>
      </div>
      <div className="col-2">
        {isSelected && (<i className="bi bi-check-circle-fill text-success fs-4"></i>)}
      </div>
    </div>
  )
}

export default User