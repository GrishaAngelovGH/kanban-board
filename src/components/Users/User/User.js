import { forwardRef } from "react"

import { Link } from "react-router-dom"

import Button from "react-bootstrap/Button"

import boardRepository from "persistent/persistentKanbanBoardRepository"
import userRepository from "persistent/persistentUserRepository"

import Avatar from "components/Avatar"
import Tooltip from "components/Tooltip"

const User = forwardRef(function User({ id, name, image, onUpdate }, ref) {
  const handleRemove = () => {
    userRepository.deleteUser(id)
    boardRepository.removeAssignedUserFromTasks(id)
    boardRepository.removeAssignedUserFromColumns(id)
    onUpdate()
  }

  return (
    <div className="row align-items-center m-0 mt-2" ref={ref}>
      <div className="col-2">
        <Avatar user={{ image, name }} />
      </div>
      <div className="col-6">
        <p className="m-0 fw-bold">{name}</p>
      </div>
      <div className="col-4">
        <Tooltip label="View assigned tasks">
          <Link to={`/tasks/${id}`} className="btn btn-success bi bi-clipboard-check me-2"></Link>
        </Tooltip>

        <Tooltip label="Delete User">
          <Button variant="danger" className="bi bi-trash" onClick={handleRemove} />
        </Tooltip>
      </div>
    </div>
  )
})

export default User