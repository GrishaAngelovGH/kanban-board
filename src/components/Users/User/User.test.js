import { render } from "@testing-library/react"

import User from "./User"

test("should render User component", () => {
  const view = render(<User name="John Smith" image="img" selected={false} />)

  expect(view).toMatchSnapshot()
})