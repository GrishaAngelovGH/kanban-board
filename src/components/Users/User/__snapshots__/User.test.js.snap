// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Users/User component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row align-items-center mt-2"
      >
        <div
          class="col-2"
        >
          <img
            alt="assigned-user"
            class="rounded-circle"
            src="img"
            width="50"
          />
        </div>
        <div
          class="col-8"
        >
          <p
            class="m-0 fw-bold"
          >
            John Smith
          </p>
        </div>
        <div
          class="col-2"
        >
          <button
            class="btn btn-danger bi bi-trash"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row align-items-center mt-2"
    >
      <div
        class="col-2"
      >
        <img
          alt="assigned-user"
          class="rounded-circle"
          src="img"
          width="50"
        />
      </div>
      <div
        class="col-8"
      >
        <p
          class="m-0 fw-bold"
        >
          John Smith
        </p>
      </div>
      <div
        class="col-2"
      >
        <button
          class="btn btn-danger bi bi-trash"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
